flycheck.el

```diff
@@ -4183,32 +4175,31 @@ The following PROPERTIES constitute an error level:
      A face symbol denoting the face to use for messages of this
      level in the error list.  See `flycheck-list-errors'."
   (declare (indent 1))
-  (setf (get level 'flycheck-error-level) t)
-  (setf (get level 'flycheck-error-severity)
-        (or (plist-get properties :severity) 0))
-  (setf (get level 'flycheck-compilation-level)
-        (plist-get properties :compilation-level))
-  (setf (get level 'flycheck-overlay-category)
-        (plist-get properties :overlay-category))
-  (setf (get level 'flycheck-fringe-bitmaps)
-        (let ((bitmap (plist-get properties :fringe-bitmap)))
-          (if (consp bitmap) bitmap (cons bitmap bitmap))))
-  ;; Kept for compatibility
-  (setf (get level 'flycheck-fringe-bitmap-double-arrow)
-        (car (get level 'flycheck-fringe-bitmaps)))
-  (setf (get level 'flycheck-fringe-face)
-        (plist-get properties :fringe-face))
-  (setf (get level 'flycheck-margin-spec)
-        (or (plist-get properties :margin-spec)
-            (flycheck-make-margin-spec
-             flycheck-default-margin-str
-             (or (get level 'flycheck-fringe-face) 'default))))
-  (setf (get level 'flycheck-margin-continuation)
-        (flycheck-make-margin-spec
-         flycheck-default-margin-continuation-str
-         (or (get level 'flycheck-fringe-face) 'default)))
-  (setf (get level 'flycheck-error-list-face)
-        (plist-get properties :error-list-face)))
+  (cl-destructuring-bind (&key error-list-face fringe-face margin-spec
+                               compilation-level overlay-category severity
+                               ((:fringe-bitmap bitmap))
+                               &allow-other-keys)
+      properties
+    (setf (get level 'flycheck-error-level) t)
+    (setf (get level 'flycheck-error-severity) (or severity 0))
+    (setf (get level 'flycheck-compilation-level) compilation-level)
+    (setf (get level 'flycheck-overlay-category) overlay-category)
+    (setf (get level 'flycheck-fringe-bitmaps)
+          (if (consp bitmap) bitmap (cons bitmap bitmap)))
+    ;; Kept for compatibility
+    (setf (get level 'flycheck-fringe-bitmap-double-arrow)
+          (car (get level 'flycheck-fringe-bitmaps)))
+    (setf (get level 'flycheck-fringe-face) fringe-face)
+    (setf (get level 'flycheck-margin-spec)
+          (or margin-spec
+              (flycheck-make-margin-spec
+               flycheck-default-margin-str
+               (or (get level 'flycheck-fringe-face) 'default))))
+    (setf (get level 'flycheck-margin-continuation)
+          (flycheck-make-margin-spec
+           flycheck-default-margin-continuation-str
+           (or (get level 'flycheck-fringe-face) 'default)))
+    (setf (get level 'flycheck-error-list-face) error-list-face)))
```

Comment:
I'm not sure it's a win to separate the put from the get.

Reply:
Fair enough